public/app/plugins/datasource/grafana-azure-monitor-datasource/components/ResourcePicker/ResourcePicker.tsx

```diff
@@ -134,88 +106,105 @@ const ResourcePicker = ({
     onApply(internalSelectedURI);
   }, [internalSelectedURI, onApply]);
 
+  const handleSearch = useCallback(
+    async (searchWord: string) => {
+      if (!searchWord) {
+        setLoadingStatus('NotStarted');
```

Comment:
I don't think you are supposed to set the loading status to 'NotStarted' since that should be only the default state. IMO, I would just make the LoadingStatus a boolean and either check in the line 48 if the status is undefined or simply change that `useEffect` to `useEffectOnce` since it's meant to be run only once.

Reply:
So when you type in a search, we erase the old contents, and if you enter backspace so that there is no longer a search phrase, we have to reset the resource picker to have some kind of results. There are a couple of ways that I could think of trying that:
1. cache the old results in a state variable and reset them back to that (this seemed bad to me because we risk stale data)
2. we could just return whatever search gives for an empty search phrase (a flattened list of everything) that seemed confusing to me and would probably cause timeouts.
3. we could retrigger the same call we do on mount which is to get all subscriptions, and only get resource groups or resources if there was a previous selection, and then only for those opened folders.

I figure 3 seemed like a reasonable response so need some way to restart the useEffect not just on mount but when the search field goes from something to nothing, so that's what's happening here. That all said "NotStarted" is not an accurate description of this either, I renamed some stuff if you want to take a look and let me know if it seems clear and like a good approach.

Reply:
gotcha, yes I agree with 3 there but modifying the loading status to trigger a fetch seems unnecessarily complex. That use effect even already defines the function loadInitialData. You just need to call it there and here so your intention is explicit.